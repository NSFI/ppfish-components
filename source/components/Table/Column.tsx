import * as React from 'react';
import { ColumnProps } from './interface';

export default class Column<T> extends React.Component<ColumnProps<T>, React.ComponentState> {}
