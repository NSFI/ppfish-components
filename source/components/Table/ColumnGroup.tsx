import * as React from 'react';

export interface ColumnGroupProps {
  title?: React.ReactNode;
}

export default class ColumnGroup extends React.Component<ColumnGroupProps, React.ComponentState> {
  static __ANT_TABLE_COLUMN_GROUP = true;
}
