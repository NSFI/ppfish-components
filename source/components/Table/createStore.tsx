export interface Store {
  setState: (partial: Object) => void;
  getState: () => any;
  subscribe: (listener: () => void) => () => void;
}

export default function createStore(initialState: object): Store {
  let state = initialState;
  const listeners: any[] = [];

  function setState(partial: object) {
    state = {
      ...state,
      ...partial,
    };
    for (let i = 0; i < listeners.length; i++) {
      listeners[i]();
    }
  }

  function getState() {
    return state;
  }

  function subscribe(listener: () => any) {
    listeners.push(listener);

    return function unsubscribe() {
      const index = listeners.indexOf(listener);
      listeners.splice(index, 1);
    };
  }

  return {
    setState,
    getState,
    subscribe,
  };
}
