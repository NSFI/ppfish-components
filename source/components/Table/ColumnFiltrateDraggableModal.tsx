import React from 'react';
import { polyfill } from 'react-lifecycles-compat';
import Modal from '../Modal';
import Icon from '../Icon';
import ToolTip from '../Tooltip';
import Table from '../Table';
import TableDraggableHOC from './TableDraggableHOC';
import { ColumnFiltrateProps, ColumnFiltrateState } from './interface';
import ConfigConsumer from '../Config/Consumer';

const getColumnKey = column => {
  if ('colSpan' in column) {
    return column.filtrateTitle || column.title;
  } else {
    return column.key || column.dataIndex || column.filtrateTitle || column.title;
  }
};

const flatten = arr => {
  return arr.reduce(function (prev, next) {
    return prev.concat(Array.isArray(next) ? flatten(next) : next);
  }, []);
};

const getCheckedOptions = (columns, hideColumns, defaultColumns) => {
  return (
      columns
      // 去除默认列
      .filter(column => defaultColumns.findIndex(key => key === getColumnKey(column)) === -1)
      // 去除表头合并不显示的列
      .filter(column => column.colSpan !== 0)
      // 去除初始化就被隐藏的列
      .filter(column => hideColumns.findIndex(key => key === getColumnKey(column)) === -1)
      // 只留uniqueId
      .map(column => getColumnKey(column))
  );
};

const getColSpanOption = columns => {
  const colSpanOption = [];
  columns.forEach((column, index) => {
    if ('colSpan' in column && column.colSpan !== 0) {
      colSpanOption.push({
        key: column.filtrateTitle || column.title,
        value: columns
          .slice(index, index + column.colSpan)
          .map(column => column.key || column.dataIndex),
      });
    }
  });
  return colSpanOption;
};

class ColumnFiltrateDraggableModal<T> extends React.Component<ColumnFiltrateProps<T>, ColumnFiltrateState> {
  static defaultProps = {
    columns: [],
    defaultColumns: [],
    hideColumns: [],
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    const { prevProps = {} } = prevState;
    const newState: any = { prevProps: nextProps };
    //监听column个数变化，重新初始化checkedOption
    if ('columns' in nextProps && nextProps.columns.length !== prevProps.columns.length) {
      const { columns, hideColumns, defaultColumns } = nextProps;
      const option = getCheckedOptions(columns, hideColumns, defaultColumns);
      newState.checkedOption = option;
      newState.checkedOptionConfirm = option;
      newState.colSpanOption = getColSpanOption(columns);
    }
    return newState;
  }

  constructor(props: ColumnFiltrateProps<T>) {
    super(props);
    const { columns, hideColumns, defaultColumns } = this.props;
    const option = getCheckedOptions(columns, hideColumns, defaultColumns);
    this.state = {
      visible: false,
      checkedOption: option,
      checkedOptionConfirm: option,
      okButtonDisabled: true,
      prevProps: props,
      colSpanOption: getColSpanOption(columns)
    };
  }

  showModal = () => {
    this.setState({
      visible: true,
    });
  };

  handleOk = () => {
    this.setState(
      {
        visible: false,
        checkedOptionConfirm: this.state.checkedOption,
        okButtonDisabled: true,
      },
      () => {
        // 被隐藏的列表key
        this.props.onChange(
          flatten(
            this.getOptions()
              .filter(column => this.state.checkedOptionConfirm.indexOf(column.value) === -1)
              .map(column => {
                const indexOfColSpan = this.state.colSpanOption.findIndex(
                  option => option.key === column.value,
                );
                if (indexOfColSpan !== -1) {
                  return this.state.colSpanOption[indexOfColSpan].value;
                } else {
                  return column.value;
                }
              }),
          ),
          this.props.defaultColumns.concat(this.state.checkedOption)
        );
      },
    );
  };

  handleCancel = () => {
    this.setState({
      visible: false,
      checkedOption: this.state.checkedOptionConfirm,
      okButtonDisabled: true,
    });
  };

  onChange = checkedOption => {
    this.setState({
      checkedOption,
      okButtonDisabled: false
    });
  };

  getOptions = () => {
    const { columns, defaultColumns } = this.props;
    return (
      columns
        // 去除表头合并的不显示的
        .filter(column => column.colSpan !== 0)
        // 获取他们的label、value、disabled
        .map(column => {
          const uniqKey = getColumnKey(column);
          //fixed /分组不能控制显示隐藏
          const disabled = !!column.fixed || !!column.children || !!column.filtrateDefault;
          const title = column.filtrateTitle || column.title;
          return {
            label: <ToolTip title={title}>{title}</ToolTip>,
            value: uniqKey,
            disabled: disabled,
          };
        })
        // 去除默认展示数据项
        .filter(column => defaultColumns.findIndex(key => key === column.value) === -1)
    );
  };

  getDataSource = () => {
    const { columns, defaultColumns } = this.props;
    const { checkedOption } = this.state;
    let selectRows = columns.filter(column => defaultColumns.findIndex(key => key === getColumnKey(column)) != -1);
    let unSelectRows = [];
    for (let key of checkedOption) {
      selectRows = selectRows.concat(columns.filter(each => each.key == key));
    }

    for (let each of columns.filter(column => defaultColumns.findIndex(key => key === getColumnKey(column)) === -1)) {
      if (checkedOption.includes(each.key)) {
        each.selected = true;
      } else {
        each.selected = false;
        unSelectRows.push(each);
      }
    }
    return selectRows.concat(unSelectRows);
  }

  render() {
    const { prefixCls, defaultColumns } = this.props;
    const { visible, checkedOption, okButtonDisabled } = this.state;
    const DraggableTable = TableDraggableHOC(Table, {
      dnd: true,
      renderBodyRow(props) {
        const { index, ...restProps } = props;
        return <tr {...restProps} />;
      },
      dndRow(props) {
        if (props.record.selected) {
          return true;
        } else {
          return false;
        }
      }
    });
    const draggableTableColumns = [
      {
        title: '字段',
        dataIndex: 'title',
        render: (val, row) => {
          return (
            <span>
              {val}
            </span>
          );
        }
      },
      {
        title: '拖拽排序',
        dataIndex: 'operate',
        render: (val, row) => {
          let isDefault = defaultColumns.findIndex(key => key === getColumnKey(row)) != -1;
          if (checkedOption.findIndex(key => key === getColumnKey(row)) != -1 && !isDefault) {
            return <Icon type="sound-drag" />
          } else {
            return null
          }
        }
      },
    ];
    return (
      <ConfigConsumer componentName="Table">
        {Locale => {
          return (
            <div>
              <Icon
                className={`${prefixCls}-filtrate-icon`}
                type="Settingx"
                onClick={this.showModal}
              />
              <Modal
                title={Locale.modalTitle}
                wrapClassName={`${prefixCls}-filtrate-modal`}
                visible={visible}
                width={680}
                okButtonDisabled={okButtonDisabled}
                onOk={this.handleOk}
                onCancel={this.handleCancel}
                okText={Locale.okText}
                cancelText={Locale.cancelText}
              >
                <DraggableTable
                  rowKey='key'
                  columns={draggableTableColumns}
                  dataSource={this.getDataSource()}
                  pagination={false}
                  rowSelection={{
                    selectedRowKeys: checkedOption,
                    hideDefaultSelections: true,
                    showSelectAll: true,
                    type: 'checkbox',
                    onChange: this.onChange,
                    getCheckboxProps: (record) => {
                      let isDefault = defaultColumns.findIndex(key => key === getColumnKey(record)) != -1;
                      return ({
                        disabled: isDefault,
                        name: record.title,
                        checked: isDefault || checkedOption.findIndex(key => key === getColumnKey(record)) != -1
                      })
                    }
                  }}
                  onRowMoved={(fields) => {
                    const checkedOption = fields.filter(each => each.selected).map(column => getColumnKey(column));
                    this.onChange(checkedOption);
                  }}
                />
              </Modal>
            </div>
          );
        }}
      </ConfigConsumer>
    );
  }
}

polyfill(ColumnFiltrateDraggableModal);

export default ColumnFiltrateDraggableModal;