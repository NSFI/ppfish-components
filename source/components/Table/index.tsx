import Table from './Table';

export * from './interface';

export default Table;
