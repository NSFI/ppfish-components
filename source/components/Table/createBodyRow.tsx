import * as React from 'react';
import classnames from 'classnames';
import omit from 'omit.js';
import { Store } from './createStore';

interface BodyRowProps {
  store: Store;
  className?: string;
  rowKey: string;
  prefixCls: string;
}

interface BodyRowState {
  selected: boolean;
}

export interface BodyRowClass extends React.ComponentClass {}

export default function createTableRow(Component = 'tr') {
  class BodyRow extends React.Component<BodyRowProps, BodyRowState> {
    private store: Store;
    private unsubscribe: () => void;

    constructor(props: BodyRowProps) {
      super(props);

      this.store = props.store;
      const { selectedRowKeys } = this.store.getState();

      this.state = {
        selected: selectedRowKeys.indexOf(props.rowKey) >= 0,
      };
    }

    componentDidMount() {
      this.subscribe();
    }

    componentWillUnmount() {
      if (this.unsubscribe) {
        this.unsubscribe();
      }
    }

    subscribe() {
      const { store, rowKey } = this.props;
      this.unsubscribe = store.subscribe(() => {
        const { selectedRowKeys } = this.store.getState();
        const selected = selectedRowKeys.indexOf(rowKey) >= 0;
        if (selected !== this.state.selected) {
          this.setState({ selected });
        }
      });
    }

    render() {
      const rowProps = omit(this.props, ['prefixCls', 'rowKey', 'store']);
      const className = classnames(
        this.props.className,
        {
          [`${this.props.prefixCls}-row-selected`]: this.state.selected,
        },
      );

      return (
        <Component {...rowProps} className={className}>
          {this.props.children}
        </Component>
      );
    }
  }

  return BodyRow as BodyRowClass;
}
