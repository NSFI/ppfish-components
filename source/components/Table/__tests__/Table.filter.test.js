import React from 'react';
import { render, mount } from 'enzyme';
import Table from '../index.tsx';

describe('Table.filter', () => {
  const filterFn = (value, record) => record.name.indexOf(value) !== -1;
  const column = {
    title: 'Name',
    dataIndex: 'name',
    filters: [
      { text: 'Boy', value: 'boy' },
      { text: 'Girl', value: 'girl' },
      {
        text: 'Title',
        value: 'title',
        children: [
          { text: 'Designer', value: 'designer' },
          { text: 'Coder', value: 'coder' },
        ],
      },
    ],
    onFilter: filterFn,
  };

  const data = [
    { key: 0, name: 'Jack' },
    { key: 1, name: 'Lucy' },
    { key: 2, name: 'Tom' },
    { key: 3, name: 'Jerry' },
  ];

  function createTable(props) {
    return <Table columns={[column]} dataSource={data} pagination={false} {...props} />;
  }

  function renderedNames(wrapper) {
    return wrapper.find('TableRow').map(row => row.props().record.name);
  }

  it('renders filter correctly', () => {
    const wrapper = render(createTable());

    expect(wrapper).toMatchSnapshot();
  });

  // it('renders menu correctly', () => {
  //   const wrapper = mount(createTable());
  //   const dropdownWrapper = render(wrapper.find('Trigger').instance().getComponent());
  //   expect(dropdownWrapper).toMatchSnapshot();
  // });
  //
  // it('renders radio filter correctly', () => {
  //   const wrapper = mount(createTable({
  //     columns: [{
  //       ...column,
  //       filterMultiple: false,
  //     }],
  //   }));
  //   const dropdownWrapper = render(wrapper.find('Trigger').instance().getComponent());
  //   expect(dropdownWrapper).toMatchSnapshot();
  // });

  it('renders custom content correctly', () => {
    jest.useFakeTimers();
    const filter = <div className="custom-filter-dropdown">custom filter</div>;
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filterDropdown: filter,
          },
        ],
      }),
    );

    // open
    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');

    jest.runAllTimers();
    wrapper.update();

    expect(wrapper.find('Trigger')).toMatchSnapshot();
  });

  it('override custom filter correctly', () => {
    /* eslint-disable-next-line */
    const filterDropdown = ({ prefixCls, setSelectedKeys, confirm, clearFilters }) => {
      return (
        <div className={`${prefixCls}-view`} id="customFilter">
          <span onClick={() => setSelectedKeys([42])} id="setSelectedKeys">
            setSelectedKeys
          </span>
          <span onClick={() => confirm()} id="confirm">
            Confirm
          </span>
          <span onClick={() => clearFilters()} id="reset">
            Reset
          </span>
        </div>
      );
    };

    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filterDropdown,
          },
        ],
      }),
    );

    const filterMenu = wrapper.find('FilterMenu').instance();

    // check if renderer well
    wrapper.find('i.fishd-dropdown-trigger').simulate('click');
    expect(wrapper.find('#customFilter')).toMatchSnapshot();

    // try to use reset btn
    expect(filterMenu.state.selectedKeys).toEqual([]);
    wrapper.find('#setSelectedKeys').simulate('click');
    expect(filterMenu.state.selectedKeys).toEqual([42]);
    wrapper.find('#reset').simulate('click');
    expect(filterMenu.state.selectedKeys).toEqual([]);

    // try to use confirm btn
    wrapper.find('i.fishd-dropdown-trigger').simulate('click');
    wrapper.find('#setSelectedKeys').simulate('click');
    expect(filterMenu.state.visible).toBe(true);
    wrapper.find('#confirm').simulate('click');
    expect(filterMenu.state.selectedKeys).toEqual([42]);
    expect(filterMenu.state.visible).toBe(false);
  });

  it('can be controlled by filterDropdownVisible', () => {
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filterDropdownVisible: true,
          },
        ],
      }),
    );

    let dropdown = wrapper.find('Dropdown').first();
    expect(dropdown.props().visible).toBe(true);

    wrapper.setProps({
      columns: [
        {
          ...column,
          filterDropdownVisible: false,
        },
      ],
    });

    dropdown = wrapper.find('Dropdown').first();
    expect(dropdown.props().visible).toBe(false);
  });

  it('if the filter is visible it should ignore the selectedKeys changes', () => {
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filterDropdownVisible: true,
          },
        ],
      }),
    );

    const filterMenu = wrapper.find('FilterMenu').instance();
    expect(filterMenu.state.selectedKeys).toEqual([]);
    wrapper.find('FilterMenu').find('input[type="checkbox"]').first().simulate('click');
    expect(filterMenu.state.selectedKeys).toEqual(['boy']);
    wrapper.setProps({ dataSource: [...data, { key: 999, name: 'Chris' }] });
    expect(filterMenu.state.selectedKeys).toEqual(['boy']);
  });

  it('fires change event when visible change', () => {
    const handleChange = jest.fn();
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            onFilterDropdownVisibleChange: handleChange,
          },
        ],
      }),
    );

    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');

    expect(handleChange).toBeCalledWith(true);
  });

  it('can be controlled by filteredValue', () => {
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filteredValue: ['Lucy'],
          },
        ],
      }),
    );

    expect(wrapper.find('tbody tr').length).toBe(1);
    wrapper.setProps({
      columns: [
        {
          ...column,
          filteredValue: [],
        },
      ],
    });
    expect(wrapper.find('tbody tr').length).toBe(4);
  });

  it('can be controlled by filteredValue null', () => {
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filteredValue: ['Lucy'],
          },
        ],
      }),
    );

    expect(wrapper.find('tbody tr').length).toBe(1);
    wrapper.setProps({
      columns: [
        {
          ...column,
          filteredValue: null,
        },
      ],
    });
    expect(wrapper.find('tbody tr').length).toBe(4);
  });

  // it('fires change event', () => {
  //   const handleChange = jest.fn();
  //   const wrapper = mount(createTable({onChange: handleChange}));
  //   const dropdownWrapper = mount(wrapper.find('Trigger').instance().getComponent());
  //
  //   dropdownWrapper.find('MenuItem').first().simulate('click');
  //   dropdownWrapper.find('.confirm').simulate('click');
  //
  //   expect(handleChange).toBeCalledWith({}, {name: ['boy']}, {});
  // });
  //
  // it('three levels menu', () => {
  //   const filters = [
  //     {text: 'Upper', value: 'Upper'},
  //     {text: 'Lower', value: 'Lower'},
  //     {
  //       text: 'Level2',
  //       value: 'Level2',
  //       children: [
  //         {text: 'Large', value: 'Large'},
  //         {text: 'Small', value: 'Small'},
  //         {
  //           text: 'Level3',
  //           value: 'Level3',
  //           children: [
  //             {text: 'Black', value: 'Black'},
  //             {text: 'White', value: 'White'},
  //             {text: 'Jack', value: 'Jack'},
  //           ],
  //         },
  //       ],
  //     },
  //   ];
  //   const wrapper = mount(createTable({
  //     columns: [{
  //       ...column,
  //       filters,
  //     }],
  //   }));
  //   jest.useFakeTimers();
  //   const dropdownWrapper = mount(wrapper.find('Trigger').instance().getComponent());
  //   dropdownWrapper.find('.fishd-dropdown-menu-submenu-title').at(0).simulate('mouseEnter');
  //   jest.runAllTimers();
  //   dropdownWrapper.update();
  //   dropdownWrapper.find('.fishd-dropdown-menu-submenu-title').at(1).simulate('mouseEnter');
  //   jest.runAllTimers();
  //   dropdownWrapper.update();
  //   dropdownWrapper.find('MenuItem').last().simulate('click');
  //   dropdownWrapper.find('.confirm').simulate('click');
  //   wrapper.update();
  //   expect(renderedNames(wrapper)).toEqual(['Jack']);
  //   jest.useRealTimers();
  // });

  it('works with JSX in controlled mode', () => {
    jest.useFakeTimers();
    const { Column } = Table;

    class App extends React.Component {
      state = {
        filters: {},
      };

      handleChange = (pagination, filters) => {
        this.setState({ filters });
      };

      render() {
        const { filters } = this.state;
        return (
          <Table dataSource={data} onChange={this.handleChange}>
            <Column
              title="name"
              dataIndex="name"
              key="name"
              filters={[
                { text: 'Jack', value: 'Jack' },
                { text: 'Lucy', value: 'Lucy' },
              ]}
              filteredValue={filters.name}
              onFilter={filterFn}
            />
          </Table>
        );
      }
    }

    const wrapper = mount(<App />);

    // Open
    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');

    jest.runAllTimers();
    wrapper.update();

    wrapper.find('MenuItem').first().simulate('click');
    wrapper.find('.confirm').simulate('click');
    wrapper.update();
    expect(renderedNames(wrapper)).toEqual(['Jack']);

    wrapper.find('.clear').simulate('click');
    wrapper.update();
    expect(renderedNames(wrapper)).toEqual(['Jack', 'Lucy', 'Tom', 'Jerry']);
  });

  it('works with grouping columns in controlled mode', () => {
    const columns = [
      {
        title: 'group',
        key: 'group',
        children: [
          {
            title: 'Name',
            dataIndex: 'name',
            key: 'name',
            filters: [
              { text: 'Jack', value: 'Jack' },
              { text: 'Lucy', value: 'Lucy' },
            ],
            onFilter: filterFn,
            filteredValue: ['Jack'],
          },
          {
            title: 'Age',
            dataIndex: 'age',
            key: 'age',
          },
        ],
      },
    ];
    const testData = [
      { key: 0, name: 'Jack', age: 11 },
      { key: 1, name: 'Lucy', age: 20 },
      { key: 2, name: 'Tom', age: 21 },
      { key: 3, name: 'Jerry', age: 22 },
    ];
    const wrapper = mount(<Table columns={columns} dataSource={testData} />);

    expect(renderedNames(wrapper)).toEqual(['Jack']);
  });

  it('confirm filter when dropdown hidden', () => {
    const handleChange = jest.fn();
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filters: [
              { text: 'Jack', value: 'Jack' },
              { text: 'Lucy', value: 'Lucy' },
            ],
          },
        ],
        onChange: handleChange,
      }),
    );

    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');
    wrapper.find('.fishd-dropdown-menu-item').first().simulate('click');
    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');

    expect(handleChange).toBeCalled();
  });

  it('renders custom filter icon correctly', () => {
    const filterIcon = filtered => <span>{filtered ? 'filtered' : 'unfiltered'}</span>;
    const wrapper = mount(
      createTable({
        columns: [
          {
            ...column,
            filterIcon,
          },
        ],
      }),
    );

    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');
    wrapper.find('.fishd-dropdown-menu-item').first().simulate('click');
    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');
    expect(wrapper.find('.fishd-table-filter-icon').render()).toMatchSnapshot();

    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');
    wrapper.find('.fishd-dropdown-menu-item').first().simulate('click');
    wrapper.find('.fishd-dropdown-trigger').first().simulate('click');
    expect(wrapper.find('.fishd-table-filter-icon').render()).toMatchSnapshot();
  });
});
