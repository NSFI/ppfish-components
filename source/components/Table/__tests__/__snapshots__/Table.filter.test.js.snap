// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.filter override custom filter correctly 1`] = `
<div
  className="fishd-dropdown-custom-view"
  id="customFilter"
>
  <span
    id="setSelectedKeys"
    onClick={[Function]}
  >
    setSelectedKeys
  </span>
  <span
    id="confirm"
    onClick={[Function]}
  >
    Confirm
  </span>
  <span
    id="reset"
    onClick={[Function]}
  >
    Reset
  </span>
</div>
`;

exports[`Table.filter renders custom content correctly 1`] = `
<Trigger
  action={
    Array [
      "click",
    ]
  }
  afterPopupVisibleChange={[Function]}
  autoDestroy={false}
  blurDelay={0.15}
  builtinPlacements={
    Object {
      "bottomCenter": Object {
        "offset": Array [
          0,
          8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "tc",
          "bc",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
      "bottomLeft": Object {
        "offset": Array [
          0,
          8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "tl",
          "bl",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
      "bottomRight": Object {
        "offset": Array [
          0,
          8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "tr",
          "br",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
      "topCenter": Object {
        "offset": Array [
          0,
          -8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "bc",
          "tc",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
      "topLeft": Object {
        "offset": Array [
          0,
          -8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "bl",
          "tl",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
      "topRight": Object {
        "offset": Array [
          0,
          -8,
        ],
        "overflow": Object {
          "adjustX": 1,
          "adjustY": 1,
        },
        "points": Array [
          "br",
          "tr",
        ],
        "targetOffset": Array [
          0,
          0,
        ],
      },
    }
  }
  defaultPopupVisible={false}
  defaultVisible={false}
  destroyPopupOnHide={false}
  focusDelay={0}
  getDocument={[Function]}
  getPopupClassNameFromAlign={[Function]}
  getPopupContainer={[Function]}
  hideAction={Array []}
  mask={false}
  maskClosable={true}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onPopupAlign={[Function]}
  onPopupVisibleChange={[Function]}
  onVisibleChange={[Function]}
  overlay={
    <FilterDropdownMenuWrapper
      focusable={true}
      mode="vertical"
      selectable={false}
    >
      <div
        className="custom-filter-dropdown"
      >
        custom filter
      </div>
    </FilterDropdownMenuWrapper>
  }
  popup={
    <FilterDropdownMenuWrapper
      focusable={true}
      mode="vertical"
      onClick={[Function]}
      prefixCls="fishd-dropdown-menu"
      selectable={false}
    >
      <div
        className="custom-filter-dropdown"
      >
        custom filter
      </div>
    </FilterDropdownMenuWrapper>
  }
  popupAlign={Object {}}
  popupClassName=""
  popupPlacement="bottomLeft"
  popupStyle={Object {}}
  popupTransitionName="slide-up"
  popupVisible={true}
  prefixCls="fishd-dropdown"
  showAction={Array []}
  visible={true}
>
  <Icon
    className="fishd-dropdown-trigger"
    key="trigger"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    title="筛选"
    type="filter"
  >
    <i
      className="fishdicon fishdicon-filter fishd-dropdown-trigger"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      title="筛选"
    />
  </Icon>
  <ForwardRef
    didUpdate={[Function]}
    getContainer={[Function]}
    key="portal"
  >
    <Portal
      containerInfo={
        <div
          style="position: absolute; top: 0px; left: 0px; width: 100%;"
        >
          <div>
            <div
              class="fishd-dropdown"
              style="opacity: 0; pointer-events: none;"
            >
              <div>
                <div
                  class="custom-filter-dropdown"
                >
                  custom filter
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <Popup
        align={
          Object {
            "offset": Array [
              0,
              8,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          }
        }
        className=""
        destroyPopupOnHide={false}
        getClassNameFromAlign={[Function]}
        getRootDomNode={[Function]}
        mask={false}
        onAlign={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        prefixCls="fishd-dropdown"
        style={Object {}}
        transitionName="slide-up"
        visible={true}
      >
        <div>
          <Mask
            align={
              Object {
                "offset": Array [
                  0,
                  8,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              }
            }
            className=""
            destroyPopupOnHide={false}
            getClassNameFromAlign={[Function]}
            getRootDomNode={[Function]}
            mask={false}
            onAlign={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            prefixCls="fishd-dropdown"
            style={Object {}}
            transitionName="slide-up"
            visible={true}
          />
          <PopupInner
            align={
              Object {
                "offset": Array [
                  0,
                  8,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              }
            }
            className=""
            destroyPopupOnHide={false}
            getClassNameFromAlign={[Function]}
            getRootDomNode={[Function]}
            mask={false}
            onAlign={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            prefixCls="fishd-dropdown"
            style={Object {}}
            transitionName="slide-up"
            visible={true}
          >
            <CSSMotion
              leavedClassName="fishd-dropdown-hidden"
              motionName="slide-up"
              onAppearEnd={[Function]}
              onAppearPrepare={[Function]}
              onEnterEnd={[Function]}
              onEnterPrepare={[Function]}
              onLeaveEnd={[Function]}
              removeOnLeave={false}
              visible={true}
            >
              <DomWrapper>
                <Align
                  align={
                    Object {
                      "offset": Array [
                        0,
                        8,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    }
                  }
                  disabled={false}
                  key="popup"
                  monitorWindowResize={true}
                  onAlign={[Function]}
                  target={[Function]}
                >
                  <div
                    className="fishd-dropdown"
                    onMouseDownCapture={[Function]}
                    onTouchStartCapture={[Function]}
                    style={
                      Object {
                        "opacity": 0,
                        "pointerEvents": "none",
                        "zIndex": undefined,
                      }
                    }
                  >
                    <FilterDropdownMenuWrapper
                      focusable={true}
                      mode="vertical"
                      onClick={[Function]}
                      prefixCls="fishd-dropdown-menu"
                      selectable={false}
                    >
                      <div
                        onClick={[Function]}
                      >
                        <div
                          className="custom-filter-dropdown"
                        >
                          custom filter
                        </div>
                      </div>
                    </FilterDropdownMenuWrapper>
                  </div>
                </Align>
              </DomWrapper>
            </CSSMotion>
          </PopupInner>
        </div>
      </Popup>
    </Portal>
  </ForwardRef>
</Trigger>
`;

exports[`Table.filter renders custom filter icon correctly 1`] = `
<span
  class="fishd-table-filter-icon fishd-dropdown-trigger"
  title="筛选"
>
  filtered
</span>
`;

exports[`Table.filter renders custom filter icon correctly 2`] = `
<span
  class="fishd-table-filter-icon fishd-dropdown-trigger"
  title="筛选"
>
  unfiltered
</span>
`;

exports[`Table.filter renders filter correctly 1`] = `
<div
  class="fishd-table-wrapper"
>
  <div
    class="fishd-spin-nested-loading"
  >
    <div
      class="fishd-spin-nested"
    >
      <div>
        <div
          class="fishd-table fishd-table-default fishd-table-scroll-position-left"
        >
          <div
            class="fishd-table-content"
          >
            <div
              class="fishd-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col />
                </colgroup>
                <thead
                  class="fishd-table-thead"
                >
                  <tr>
                    <th
                      class="fishd-table-column-has-filters"
                    >
                      <span>
                        Name
                        <i
                          class="fishdicon fishdicon-filter fishd-dropdown-trigger"
                          title="筛选"
                        />
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="fishd-table-tbody"
                >
                  <tr
                    class="fishd-table-row fishd-table-row-level-0"
                    data-row-key="0"
                  >
                    <td
                      class="fishd-table-column-has-filters"
                    >
                      <span
                        class="fishd-table-row-indent indent-level-0"
                        style="padding-left:0px"
                      />
                      Jack
                    </td>
                  </tr>
                  <tr
                    class="fishd-table-row fishd-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class="fishd-table-column-has-filters"
                    >
                      <span
                        class="fishd-table-row-indent indent-level-0"
                        style="padding-left:0px"
                      />
                      Lucy
                    </td>
                  </tr>
                  <tr
                    class="fishd-table-row fishd-table-row-level-0"
                    data-row-key="2"
                  >
                    <td
                      class="fishd-table-column-has-filters"
                    >
                      <span
                        class="fishd-table-row-indent indent-level-0"
                        style="padding-left:0px"
                      />
                      Tom
                    </td>
                  </tr>
                  <tr
                    class="fishd-table-row fishd-table-row-level-0"
                    data-row-key="3"
                  >
                    <td
                      class="fishd-table-column-has-filters"
                    >
                      <span
                        class="fishd-table-row-indent indent-level-0"
                        style="padding-left:0px"
                      />
                      Jerry
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
